### ISSUE TITLE
`VoxelMeshSDF._set_data` crashes Godot

### ISSUE THREAD

**Issue by qarmin** [Zylann/godot_voxel#463]
```
extends Node
func _process(delta):

	var temp_variable714 = VoxelMeshSDF.new()
	temp_variable714._set_data({"roman" : 22, 22 : 25, BoxShape3D.new() : BoxShape3D.new()})
```
crashes with backtrace
```
WARNING: VoxelBuffer::create called with empty size (0, 0, 0)
     at: create (modules/voxel/storage/voxel_buffer_internal.cpp:171)
ERROR: FATAL: Assertion failed: "size != 0" is false.
   at: allocate (modules/voxel/storage/voxel_memory_pool.cpp:95)

================================================================
handle_crash: Program crashed with signal 4
Engine version: Godot Engine v4.0.beta.custom_build (0e9878b12a1e91d0f9ce2c1b56d1d469dd1758f6)
Dumping the backtrace. Please include this when reporting the bug on: https://github.com/godotengine/godot/issues
[1] /mnt/Miecz/godot/bin/godot.linuxbsd.editor.dev.x86_64.san(+0x3e92b07c) [0x5621e58c907c] (??:0)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7f18318fa520] (??:0)
[3] zylann::voxel::VoxelMemoryPool::allocate(unsigned long) (??:0)
[4] zylann::voxel::allocate_channel_data(unsigned long) (??:0)
[5] zylann::voxel::VoxelBufferInternal::create_channel_noinit(int, Vector3i) (??:0)
[6] zylann::voxel::VoxelBufferInternal::create_channel(int, unsigned long) (??:0)
[7] zylann::voxel::VoxelBufferInternal::decompress_channel(unsigned int) (??:0)
[8] zylann::voxel::VoxelMeshSDF::_b_set_data(Dictionary) (??:0)
[9] void call_with_variant_args_helper<__UnexistingClass, Dictionary, 0ul>(__UnexistingClass*, void (__UnexistingClass::*)(Dictionary), Variant const**, Callable::CallError&, IndexSequence<0ul>) (??:0)
[10] void call_with_variant_args_dv<__UnexistingClass, Dictionary>(__UnexistingClass*, void (__UnexistingClass::*)(Dictionary), Variant const**, int, Callable::CallError&, Vector<Variant> const&) (??:0)
[11] MethodBindT<Dictionary>::call(Object*, Variant const**, int, Callable::CallError&) (??:0)
[12] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (??:0)
[13] Variant::callp(StringName const&, Variant const**, int, Variant&, Callable::CallError&) (??:0)
[14] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (??:0)
[15] GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) (??:0)
[16] bool Node::_gdvirtual__process_call<false>(double) (??:0)
[17] Node::_notification(int) (??:0)
[18] Node::_notificationv(int, bool) (??:0)
[19] Object::notification(int, bool) (??:0)
[20] SceneTree::_notify_group_pause(StringName const&, int) (??:0)
[21] SceneTree::process(double) (??:0)
[22] Main::iteration() (??:0)
[23] OS_LinuxBSD::run() (??:0)
[24] /mnt/Miecz/godot/bin/godot.linuxbsd.editor.dev.x86_64.san(main+0x583) [0x5621e58c62fc] (??:0)
[25] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7f18318e1d90] (??:0)
[26] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7f18318e1e40] (??:0)
[27] /mnt/Miecz/godot/bin/godot.linuxbsd.editor.dev.x86_64.san(_start+0x25) [0x5621e58c5cb5] (??:0)
-- END OF BACKTRACE --
================================================================
Aborted (core dumped)
```

**Comment by Zylann** [comment]
Should be fixed by 77156924c650500b5ebfe1780e13669bdeece9ac

**Comment by qarmin** [comment]
Still have crash with same code

**Comment by Zylann** [comment]
More fixes in 5dcda4aaa75c122da524d95639859ecbf8c3334b

**Comment by Zylann** [comment]
No longer happening in the fuzzer, closing.
